### ISSUE TITLE
how to custom dataset?

### ISSUE THREAD

**Issue by kimtaehyeong** [qubvel/segmentation_models.pytorch#150]
Hello.
I am trying to train pytorch based efficientunet code with my data.
I'm currently trying to learn 'cars segmentation (camvid)' with this example code, and my dataset is data consisting of (rgb, depth). I have a data set without a label and I'm testing it with only CLASSES = ['depth']. I'm constantly getting a purple background from a mask image, and I want to know how to modify the code to learn from an image consisting of (rgb, depth).
Thank you

**Comment by kimtaehyeong** [comment]
Ultimately it seems like I could modify the Dataset class, but did not get the desired result.
How can I fix it?
Data configuration is composed of NYU dataset, and made of 3 trains, vals, and tests as in the example folder.
thanks :)
ex)
![9090](https://user-images.githubusercontent.com/11425786/75112850-4f5a9200-568b-11ea-80e6-61dfd807492a.PNG)

ex class)
class Dataset(BaseDataset):
    """CamVid Dataset. Read images, apply augmentation and preprocessing transformations.
    
    Args:
        images_dir (str): path to images folder
        masks_dir (str): path to segmentation masks folder
        class_values (list): values of classes to extract from segmentation mask
        augmentation (albumentations.Compose): data transfromation pipeline 
            (e.g. flip, scale, etc.)
        preprocessing (albumentations.Compose): data preprocessing 
            (e.g. noralization, shape manipulation, etc.)
    
    """
    
    CLASSES = ['depth'] # do not have a class.
    
    def __init__(
            self, 
            images_dir, 
            classes=None, 
            augmentation=None, 
            preprocessing=None,
    ):
        self.ids = os.listdir(images_dir)
        self.images_fps = [os.path.join(images_dir, image_id) for image_id in self.ids]
        self.masks_fps = [os.path.join(masks_dir, image_id) for image_id in self.ids]
        
        # convert str names to class values on masks
        self.class_values = [self.CLASSES.index(cls.lower()) for cls in classes]
        
        self.augmentation = augmentation
        self.preprocessing = preprocessing
    
    def __getitem__(self, i):
        
        # read data
        image = cv2.imread(self.images_fps[i])
        image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
        mask = cv2.imread(self.masks_fps[i], 0)
        
        # extract certain classes from mask (e.g. cars)
        masks = [(mask == v) for v in self.class_values]
        mask = np.stack(masks, axis=-1).astype('float')
        
        # apply augmentations
        if self.augmentation:
            sample = self.augmentation(image=image, mask=mask)
            image, mask = sample['image'], sample['mask']
        
        # apply preprocessing
        if self.preprocessing:
            sample = self.preprocessing(image=image, mask=mask)
            image, mask = sample['image'], sample['mask']
            
        return image, mask
        
    def __len__(self):
        return len(self.ids)

**Comment by suitre77** [comment]
a segmentation label based on class id's for every pixel as integer values. for instance one class has a unique integer. from this ground truth different masks can be extracted as one-hot-encoded matrices. it is inconvenient to map a depth label to segmentation masks. to that depth estimation is another task than segmentation
